database/redis/reply/delivery_channel.go

```diff
@@ -0,0 +1,51 @@
+package reply
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/moira-alert/moira/database"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/moira-alert/moira"
+)
+
+func unmarshalDeliveryChannel(bytes []byte, err error) (moira.DeliveryChannel, error) {
+	deliveryChannel := moira.DeliveryChannel{}
+	if err != nil {
+		if errors.Is(err, redis.Nil) {
+			return deliveryChannel, database.ErrNil
+		}
+		return deliveryChannel, fmt.Errorf("failed to read contact: %s", err.Error())
+	}
+
+	err = json.Unmarshal(bytes, &deliveryChannel)
+	if err != nil {
+		return deliveryChannel, fmt.Errorf("failed to parse delivery channel json %s: %s", string(bytes), err.Error())
```

Comment:
todo: fmt.Errorf("failed to parse delivery channel json %s: %w", string(bytes), err)